src/NHibernate/Impl/AbstractQueryImpl.cs

```diff
@@ -674,8 +676,33 @@ public IQuery SetProperties(IDictionary map)
 			return this;
 		}
 
+		private IQuery SetParameters(IDictionary<string, object> map)
+		{
+			foreach (var namedParam in NamedParameters)
+			{
+				var obj = map[namedParam];
+				switch (obj)
+				{
+					case null:
+						break;
+					case IEnumerable enumerable when !(enumerable is string):
+						SetParameterList(namedParam, enumerable);
+						break;
+					default:
+						SetParameter(namedParam, obj, DetermineType(namedParam, obj.GetType()));
+						break;
+				}
+			}
+			return this;
+		}
+
 		public IQuery SetProperties(object bean)
 		{
+			if (bean is IDictionary<string, object> map)
+			{
+				return SetParameters(map);
+			}
+
 			System.Type clazz = bean.GetType();
```

Comment:
Prob add call to non generic IDictionary as well

Reply:
Do you want me to add support for the non generic dictionary?

It could be done, but I do not think this will be actually useful for most users. I think only very old pieces of code may benefit of it, and well, we could leave it to the user to migrate or convert to more modern classes than the `HashTable` and the like.

I am on the side of dropping non-generic support whenever it may reduce the quantity of code we have, so I do not tend to add support for it.

I am wrong, is there still modern needs for non-generic support?

Maybe it could help those who use a generic dictionary with the "wrong" key/value types, but should we care for this case?

Reply:
IMO, generics do not make much sense when `T` is an `object`.

Reply:
They have less added value in such case, but still have quite some:
 - more consistent code: `Dictionary<keyType, valueType>` / `Dictionary<object, object>` vs `Dictionary<keyType, valueType>` / `HashTable`, `List<valueType>` / `List<object>`vs `List<valueType>` / `ArrayList`.
 - disposable `IEnumerator<valueType>` vs non-disposable `IEnumerator` (with some special handling in `for each` in order to still dispose them when the underlying concrete implementation is disposable).
 - non generic types APIs seem quite frozen (no non-generic set were added when the generic one was added, for example).
 - "object generic" collections can be fed to methods taking generic interfaces arguments, while non-generic collections require dedicated methods: using `IList<object>` / `IList<valueType>` can be handled by a single `DoSomething(IList<T>)`, while using `IList` / `IList<object>` would require an additional `DoSomething(IList)` overload.

The only added value of non-generic types vs "object generic" I see, it is more concise to declare non-generic. But their somewhat inconsistent naming nullifies this point in my view. And the potential for having to duplicate methods is a major pain.

Reply:
Marking as resolved since support for non generic `IDictionary` has been added, even though I am on the side of dropping non-generic support.